flang/include/flang/Optimizer/Transforms/Passes.td

```diff
@@ -179,4 +179,13 @@ def CharacterConversion : Pass<"character-conversion"> {
   ];
 }
 
+def ExternalNameConversion : Pass<"external-name-interop", "mlir::ModuleOp"> {
+  let summary = "Convert name for external interoperability";
+  let description = [{
+    Demangle FIR internal name and mangle them for external interoperability.
+  }];
+  let constructor = "::fir::createExternalNameConversionPass()";
+  let dependentDialects = [ "fir::FIROpsDialect", "mlir::LLVM::LLVMDialect" ];
+}
```

Comment:
I think we'll want a pass option so the user can select a mangling scheme. (Fortran compilers each have their own.)

That can be a separate PR, though. Just handling the basic case to be F77 mode compatible is a great start.

Reply:
Yeah I initially added the option and since the F77 scheme is the same for all compiler I tested I just removed it for now.